sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala

```diff
@@ -166,6 +166,30 @@ class InsertIntoHiveTableSuite extends QueryTest with TestHiveSingleton with Bef
     sql("DROP TABLE tmp_table")
   }
 
+  test("Delete the temporary staging directory and files after each insert") {
```

Comment:
let's wait for https://github.com/apache/spark/pull/16104 so that we can move this test to `VersionSuite` to test all supported hive versions.